utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/encode/AvroSchemaOutputValidator.scala

```diff
@@ -0,0 +1,292 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+import cats.data.Validated.{Invalid, condNel}
+import cats.data.{NonEmptyList, Validated, ValidatedNel}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.avro.Schema.Type
+import org.apache.avro.SchemaCompatibility.SchemaCompatibilityType
+import org.apache.avro.generic.GenericData.{EnumSymbol, Fixed}
+import org.apache.avro.{LogicalTypes, Schema, SchemaCompatibility}
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.avro.AvroUtils
+import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
+import pl.touk.nussknacker.engine.util.output._
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.util.UUID
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+import scala.util.Try
+
+private[encode] case class AvroSchemaExpected(schema: Schema) extends OutputValidatorExpected {
+  override def expected: String = AvroSchemaOutputValidatorPrinter.print(schema)
+}
+
+object AvroSchemaOutputValidator {
+  private[encode] val SimpleAvroPath = "Data"
+}
+
+class AvroSchemaOutputValidator(validationMode: ValidationMode) extends LazyLogging {
+
+  import AvroSchemaOutputValidator._
+
+  import scala.collection.JavaConverters._
+
+  private val valid = Validated.Valid(())
+
+  private val longLogicalTypes = Set(
+    LogicalTypes.timeMicros(), LogicalTypes.timestampMillis(), LogicalTypes.timestampMicros()
+  )
+
+  /**
+    * see {@link pl.touk.nussknacker.engine.avro.encode.BestEffortAvroEncoder} for underlying avro types
+    */
+  def validateTypingResultToSchema(typingResult: TypingResult, parentSchema: Schema)(implicit nodeId: NodeId): ValidatedNel[OutputValidatorError, Unit] =
+    validateTypingResult(typingResult, parentSchema, None)
+
+  final private def validateTypingResult(typingResult: TypingResult, schema: Schema, path: Option[String]): ValidatedNel[OutputValidatorError, Unit] = {
+    (typingResult, schema.getType) match {
+      case (tc@TypedClass(cl, _), _) if AvroUtils.isSpecificRecord(cl) =>
+        validateSpecificRecord(tc, schema, path)
+      case (record: TypedObjectTypingResult, Type.RECORD) =>
+        validateRecordSchema(record, schema, path)
+      case (map: TypedObjectTypingResult, Type.MAP) =>
```

Comment:
Changed as we have spoken.

Reply:
I'm not sure if spoken about this. IMO if you send record with nested field ot record type in place of map, it will blow up in runtime and we should check if objType is a map

Reply:
You're right.. I've fixed it and added some test:
```
(rConfig(Map("price" -> sampleDouble.toString), fakeMapRecordWithStringPrice, recordMapOfStringsSchema, Input), invalidTypes("path 'field' actual: '{price: String}' expected: 'Map[String, Null | String]'")),
```